jetty-websocket/websocket-jetty-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketFrameHandler.java

```diff
@@ -199,7 +198,8 @@ public void onFrame(Frame frame, Callback callback)
                     break;
 
                 case SUSPENDING:
-                    delayedOnFrame = () -> onFrame(frame, callback);
+                    delayedFrame = frame;
```

Comment:
I'm so glad we are getting rid of suspend/resume!
Can you add an `assert delayedFrame==null` here just to show that we know that there can only be 1 delayed frame.